### ISSUE TITLE
Why install.py installs both opencv-python-headless and opencv-python which make conflicts with other plugins?

### ISSUE THREAD

**Issue by RageshAntonyHM** [ltdrdata/ComfyUI-Impact-Pack#395]
I have an issue.

If I install other plugins after installing Impact Pack or vice versa, I am getting startup error that throws `no module found: cv2` or `Cannot import .../ComfyUI-Impact-Pack module for custom nodes: partially initialized module 'cv2' has no attribute 'mat_wrapper' (most likely due to a circular import)`

I heard that this is due to conflicting dependencies of `OpenCV-python` libs. I found the code` install.py` installs both `opencv-python-headless `and` opencv-python`. But as per their manual one should install any one

And many plugins for ComfyUI use `opencv-python`. So `opencv-python-headless ` conflicts them.

https://github.com/ltdrdata/ComfyUI-Impact-Pack/blob/1e6e0f92bfa460c9e12a5aabc3683d97c111089e/install.py#L197-L209

Why install.py does this? 

Could you make it to install `opencv-python` alone so it won't conflict with other plugins.

**Comment by ltdrdata** [comment]
if you have installed opencv already. That step will be skipped.

**Comment by RageshAntonyHM** [comment]
@ltdrdata 

Yes. But if I install it as a first plugin, it installs the `opencv-python-headless` which conflicts with other plugins "opencv-python"

Why it installs both ?

**Comment by ltdrdata** [comment]
> @ltdrdata
> 
> Yes. But if I install it as a first plugin, it installs the `opencv-python-headless` which conflicts with other plugins "opencv-python"
> 
> Why it installs both ?

That is related to this issue.

https://github.com/ltdrdata/ComfyUI-Impact-Pack/pull/266

When issues typically arise upon multiple installation, it's not the act of multiple installation itself that's problematic. 
The issue arises because only one version gets loaded, and if the initially loaded version is incompatible, problems occur.

**Comment by RageshAntonyHM** [comment]
@ltdrdata 
Okay
How to resolve this ?

**Comment by ltdrdata** [comment]
> @ltdrdata Okay How to resolve this ?


If your OpenCV is causing conflicts, remove both `opencv-python-headless` and `opencv-python`, then install `opencv-python` along with the pip package causing the conflict. This will ensure that the installation occurs without conflicting with the version of the pip package in question.

**Comment by oxc** [comment]
I want to chime in here, it seems strange that the code is attempting to install both packages. 

The [opencv-python documentation](https://pypi.org/project/opencv-python-headless/#installation-and-usage) is quite clear on this:

> There are four different packages (see options 1, 2, 3 and 4 below) and you should **SELECT ONLY ONE OF THEM**. Do not install multiple different packages in the same environment. There is no plugin architecture: all the packages use the same namespace (cv2). If you installed multiple different packages in the same environment, uninstall them all with pip uninstall and reinstall only one package.

I suggest that the code should only try to install opencv-headless, which is in alignment with the requirements.txt

**Comment by ltdrdata** [comment]
There's a somewhat complicated situation there, but I plan to resolve it soon.
